libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Network/Network/Queue/CSFNetwork+Salesforce.m

```diff
@@ -38,20 +40,53 @@ - (void)setDefaultConnectCommunityId:(NSString *)defaultConnectCommunityId {
 
 - (void)setupSalesforceObserver {
     [[NSNotificationCenter defaultCenter] addObserver:self
-                                             selector:@selector(userAccountManagerDidChangeCurrentUser:)
-                                                 name:SFUserAccountManagerDidChangeCurrentUserNotification
+                                             selector:@selector(userAccountManagerDidChangeUserData:)
+                                                 name:SFUserAccountManagerDidChangeUserDataNotification
+                                               object:nil];
+
+    [[NSNotificationCenter defaultCenter] addObserver:self
+                                             selector:@selector(userAccountManagerDidChangeUser:)
+                                                 name:SFUserAccountManagerDidChangeUserNotification
                                                object:nil];
 }
 
 #pragma mark SFAuthenticationManagerDelegate
+- (void)userAccountManagerDidChangeUser:(NSNotification*)notification {
 
-- (void)userAccountManagerDidChangeCurrentUser:(NSNotification*)notification {
     SFUserAccountManager *accountManager = (SFUserAccountManager*)notification.object;
+    SFUserAccountChange change = (SFUserAccountChange)[notification.userInfo[SFUserAccountManagerUserChangeKey] integerValue];
+
+    if ([accountManager isKindOfClass:[SFUserAccountManager class]]
+            && (change & SFUserAccountChangeCurrentUser)) {
+
+        if (![accountManager.currentUserIdentity isEqual:self.account.accountIdentity]) {
+            self.networkSuspended = YES;
+        } else {
+            [self resetSession];
+            self.networkSuspended = NO;
+        }
+    }
+}
+
+- (void)userAccountManagerDidChangeUserData:(NSNotification*)notification {
+    SFUserAccountManager *accountManager = (SFUserAccountManager*)notification.object;
+    SFUserAccountDataChange change = (SFUserAccountDataChange)[notification.userInfo[SFUserAccountManagerUserChangeKey] integerValue];
     if ([accountManager isKindOfClass:[SFUserAccountManager class]]) {
-        if ([accountManager.currentUserIdentity isEqual:self.account.accountIdentity] &&
-            ![accountManager.currentCommunityId isEqualToString:self.defaultConnectCommunityId])
-        {
-            self.defaultConnectCommunityId = accountManager.currentCommunityId;
+        SFUserAccount *userAccount = notification.userInfo[SFUserAccountManagerUserChangeUserKey];
+        if([self.account.accountIdentity isEqual:userAccount.accountIdentity]) {
+
+            if ([accountManager.currentUserIdentity isEqual:self.account.accountIdentity] &&
```

Comment:
This part of the compound condition should not be here.  Actionability is only dependent on the notification coming from the user that matches this instance's account—which is checked immediately before this.